source/generators/SingleParticle.cc

```diff
@@ -145,9 +145,9 @@ void SingleParticle::GeneratePrimaryVertex(G4Event* event)
     pz = pmod * momentum_Z_/mom_mod;
   } else if (costheta_min_ != -1. || costheta_max_ != 1. || phi_min_ != 0. || phi_max_ !=2.*pi) {
     G4ThreeVector p = nexus::Direction(costheta_min_, costheta_max_, phi_min_, phi_max_);
-    px = p.x();
-    py = p.y();
-    pz = p.z();
+    px = p.x() * pmod;
```

Comment:
Why not use scalar-vector multiplication to scale the whole vector in one go, rather than doing it component by copmonent?

Reply:
Sure, that would be better! What do you suggest?

Reply:
Something along the lines of what I submitted in the other PR:
https://github.com/next-exp/nexus/blob/6829fcfc397f405808f21e746339bd873290b9b1/source/generators/Back2backGammas.cc#L82-L97

Furthermore, once an acceptable `cosTheta` has been found, it's probably pointlessly wasteful to throw it away if `phi` is inadequate, so there should probably be a second, inner infinite loop to find `phi` without throwing away `cosTheta`.

Reply:
On the other hand, since we need the three components individually to be used in the instantiation of `G4PrimaryParticle`, maybe it's more convenient to leave it component by component.

Reply:
That's why vectors have component accessors.

Reply:
Done.